### DISCUSSION TITLE
Jquery transpilation error

### DISCUSSION THREAD

**Discussion by coaperator** [babel/babel#12298] [Q&A]
Hello everybody!
help please solve the following problem
I am trying to compile Babel code
after compilation I get an error:
Uncaught TypeError: Cannot read property 'jQuery' of undefined

excluding jQuery from the stream (I collect it with a gulp) will not work in any way - jq is connected via gulp-include somewhere in the depths
google does not help (

I understand that something needs to be adjusted in Babel .. I tried to set strict: true, but it does not help

my config:
```
return gulp
    .src(sourcePaths)
    .pipe(
        plumber({
            errorHandler: notify.onError(err => ({
                title: `Task Failed [${err.plugin}]`,
                message: err.message,
                filename: err.filename,
                line: err.line,
            })),
        })
    )
    .pipe(order(orderList))
    .pipe(
        include({
            includePaths: [
                config.root,
                // `${__dirname}/node_modules`,
                // `${__dirname}/sources`,
            ],
        }).on('error', console.log)
    )
    .pipe(gulpIf(isDev, sourcemaps.init()))
    .pipe(
        gulpIf(
            isDev,
            babel({
                compact: false,
                presets: ['@babel/preset-env'],
                plugins: [
                    [
                        '@babel/plugin-transform-modules-commonjs',
                        {
                            strict: false,
                        },
                    ],
                ],
            })
        )
    )
    .pipe(
        gulpIf(
            !isDev,
            terser({
                keep_fnames: true,
                mangle: false,
                warnings: false,
                compress: {
                    drop_console: true,
                },
                format: {
                    comments: false,
                },
            })
        )
    )
    .pipe(concat(concatName))
    .pipe(gulpIf(isDev, sourcemaps.write('.')))
    .pipe(gulpIf(!isDev, rev()))
    .pipe(gulp.dest(destPath))
    .pipe(
        gulpIf(
            !isDev,
            rev.manifest('temp/rev-manifest.json', {
                base: 'temp',
                merge: true,
            })
        )
    )
    .pipe(gulpIf(!isDev, gulp.dest('temp')))
    .pipe(livereload());
```

**Comment by JLHwung** [comment]
Please offer a reproduction repo, otherwise it can be hard to debug. The error is likely due to `window.jQuery = (jQuery init code)` is run on a non-browser (node.js) environment.

**Reply by coaperator** [reply]
> Please offer a reproduction repo, otherwise it can be hard to debug. The error is likely due to `window.jQuery = (jQuery init code)` is run on a non-browser (node.js) environment.

sorry, my repository is private

no, I build a bundle and run it through the babel, run it in the browser
after compilation:
![image](https://user-images.githubusercontent.com/6645911/97920136-0e626380-1d6a-11eb-8f05-699e78cd3ee5.png)
![image](https://user-images.githubusercontent.com/6645911/97920181-22a66080-1d6a-11eb-8eeb-5d873380bb2b.png)
![image](https://user-images.githubusercontent.com/6645911/97920435-80d34380-1d6a-11eb-8b8b-324f913a6537.png)

**Reply by JLHwung** [reply]
As you can see `this` is `undefined`, you can try to paste the minified code to http://prettier.io/playground and figure out what is going wrong.

See [here](https://gist.github.com/Rich-Harris/88c5fc2ac6dc941b22e7996af05d70ff) for why reproducing repo is important when you report an issue, you can try to isolate your issue and constructing a minimal one. Without that repo I can't figure it out what is going wrong either.

**Comment by nyngwang** [comment]
@coaperator In what circumstance did you decide to set `strict: false` for `@babel/plugin-transform-modules-commonjs`?
